CHANGELOG.md

```diff
@@ -3,6 +3,160 @@
 All notable changes to Router will be documented in this file.
 
 This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+# [0.9.4] - 2022-06-14
+
+## ❗ BREAKING ❗
+### The `apollo-router-core` crate has been merged into `apollo-router` ([PR #1189](https://github.com/apollographql/router/pull/1189))
+
+To upgrade, remove any dependency on the `apollo-router-core` crate from your `Cargo.toml` files and change imports like so:
+
+```diff
+- use apollo_router_core::prelude::*;
++ use apollo_router::prelude::*;
+```
+
+By [@SimonSapin](https://github.com/SimonSapin) in https://github.com/apollographql/router/pull/1189
+
+
+### Fix input validation rules ([PR #1211](https://github.com/apollographql/router/pull/1211))
+The GraphQL specification provides two sets of coercion / validation rules, depending on whether we're dealing with inputs or outputs.
+We have added validation rules for specified input validations which were not previously implemented.
+This is a breaking change since slightly invalid input may have validated before but will now be guarded by newly-introduced validation rules.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/1211
+
```

Comment:
```suggestion
### Groundwork for `@defer` support ([PR #1175](https://github.com/apollographql/router/pull/1175)[PR #1206](https://github.com/apollographql/router/pull/1206))

To prepare for the implementation of the `@defer` directive, the `ExecutionResponse`  and `RouterResponse` types now carry a stream of responses instead of a unique response. For now that stream contains only one item, so there is no change in behaviour.

By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/1206
```